src/robot/output/listeners.py

```diff
@@ -221,15 +221,21 @@ class ListenerProxy(AbstractLoggerProxy):
                 'output_file', 'report_file', 'log_file', 'debug_file',
                 'xunit_file', 'close']
 
-    def __init__(self, name, args):
-        listener = self._import_listener(name, args)
-        AbstractLoggerProxy.__init__(self, listener)
-        self.name = name
-        self.version = self._get_version(listener)
+    def __init__(self, listener, args):
+        listener_instance = self._get_listener(listener, args)
+        AbstractLoggerProxy.__init__(self, listener_instance)
+        self.name = str(listener)
```

Comment:
Does this really work? Don't you just get the string representation of the listener instance which, by default, is just something like `<module.Class object at 0x7f3519d63590>`?